gatling-datadog/src/main/scala/io/gatling/datadog/DatadogRequests.scala

```diff
@@ -1,249 +1,262 @@
-package io.gatling.datadog
+/*
+ * Copyright 2011-2023 GatlingCorp (https://gatling.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 
-import java.util.UUID
+package io.gatling.datadog
 
-import scala.concurrent.{ExecutionContext, Future, blocking}
-import scala.jdk.CollectionConverters.{ConcurrentMapHasAsScala, EnumerationHasAsScala}
-import scala.util.{Failure, Success}
+import scala.concurrent.ExecutionContext
+import scala.jdk.CollectionConverters.EnumerationHasAsScala
 
-import akka.actor.{Actor, Props}
-import com.typesafe.scalalogging.Logger
+import io.gatling.core.akka.BaseActor
 import io.gatling.core.config.GatlingConfiguration
 import io.gatling.datadog.DatadogRequests.sendMetrics
+
+import akka.actor.{ Props, Stash }
+import com.typesafe.scalalogging.Logger
 import scaladog.Client
-import scaladog.api.{DatadogSite, StatusResponse}
-import scaladog.api.metrics.{MetricType, Point, Series}
+import scaladog.api.{ DatadogSite, StatusResponse }
+import scaladog.api.metrics.MetricType
 
 object DatadogRequests {
   def sendTotalStartedUsers(
       client: Client,
-      batchSize: Int,
       runId: String,
       simulation: String,
-      startedUsers: List[(UUID, UserLabels)]
-  )(implicit
-      ec: ExecutionContext
-  ): Seq[Future[StatusResponse]] = {
-    val futures = for {
-      (scenario, users) <- startedUsers.groupBy(_._2.scenario)
-      batchedUsers <- users.grouped(batchSize).toSeq
+      startedUsers: List[UserLabels]
+  ): Seq[StatusResponse] = {
+    val requests = for {
+      (scenario, users) <- startedUsers.groupBy(_.scenario)
     } yield {
-      send(
+      sendCount(
         client,
         "gatling.performance_tests.total_started_users",
-        MetricType.Gauge,
-        batchedUsers,
+        users.length,
         Seq(
           s"run:$runId",
-          s"simulation:$simulation",
-          s"scenario:$scenario"
+          s"simulation:$simulation"
         )
       )
     }
-    futures.toSeq
+    requests.toSeq
   }
 
   def sendTotalFinishedUsers(
```

Comment:
(nit) looks like these send functions are only used from within the object, the `sendMetrics` function seems to be the main public one. Generally good to put the public function at the top and private functions it uses internally below so it's clear what the external use of the object/class is